### PR TITLE
Contributing test-cases for NSOperation/NSOperationQueue.

### PR DESCRIPTION
**Pull request by mamabusi**

### CODE REVIEW

File: TestFoundation/TestNSOperationQueue.swift [hunk 1]
```diff
@@ -36,17 +41,21 @@ class TestNSOperationQueue : XCTestCase {
 
     func test_OperationPriorities() {
         var msgOperations = [String]()
-        let operation1 : BlockOperation = BlockOperation(block: {
+        let operation1 = BlockOperation(block: {
             msgOperations.append("Operation1 executed")
+            sleep(1)
```
Comment (mamabusi) [inline TestFoundation/TestNSOperationQueue.swift]: Added time-lag to resolve the test-case failures seen sometimes on TestFoundation.

Comment (parkera) [inline TestFoundation/TestNSOperationQueue.swift]: Sorry - sleep cannot be a fix for issues like this. Instead we should use a semaphore to signal completion of whatever work we are waiting on.

Comment (mamabusi) [inline TestFoundation/TestNSOperationQueue.swift]: Ok. Will modify to use semaphores. Thank you.

**Comment by erg** [comment]
The test is correct and exposed a real bug in the underlying concurrency code that is fixed by #621.

This PR is not needed now and can be closed.

**Pull request closed without merging**